identity_credential/src/domain_linkage/domain_linkage_validator.rs

```diff
@@ -59,39 +59,101 @@ impl<V: JwsVerifier> JwtDomainLinkageValidator<V> {
     domain: &Url,
     validation_options: &JwtCredentialValidationOptions,
   ) -> DomainLinkageValidationResult {
+    let (oks, errors): (Vec<_>, Vec<_>) = self
+      .validate_linkage_iter(issuer, configuration, domain, validation_options)?
+      .partition(Result::is_ok);
+
+    if !oks.is_empty() {
+      Ok(())
+    } else if !errors.is_empty() {
+      Err(DomainLinkageValidationError {
+        cause: DomainLinkageValidationErrorCause::List,
+        source: Some(
+          DomainLinkageValidationErrorList::new(
+            errors
+              .into_iter()
+              .map(|r| Result::unwrap_err(r)) // safe as errors are a list of pre-filtered errors
```

Comment:
I'm surprised clippy didn't complain about this, so I might be wrong but:
```suggestion
              .map(Result::unwrap_err) // Safety: `errors` is a list of prefiltered `Err(_)`.
```

Reply:
Hm, rust analyzer was complaining about something with the function references before, so I used a closure, but works now and looks better. :D